packages/openapi3/test/xml-models.test.ts

```diff
@@ -0,0 +1,665 @@
+          tags: Tag[];
+        }
+
+        @name("XmlTag")
+        model Tag {
+          name: string;
+        }`
+      );
+
+      deepStrictEqual(res.schemas.Tag, {
+        type: "object",
+        properties: {
+          name: {
+            type: "string",
+          },
+        },
+        required: ["name"],
+        xml: {
+          name: "XmlTag",
+        },
+      });
+
+      deepStrictEqual(res.schemas.Pet, {
+        type: "object",
+        properties: {
+          tags: {
+            type: "array",
+            xml: {
+              name: "ItemsTags",
+              wrapped: true,
+            },
+            items: {
+              $ref: "#/components/schemas/Tag",
+            },
+          },
+        },
+        required: ["tags"],
+        xml: {
+          name: "XmlPet",
+        },
+      });
+    });
+    it("Scenario 2.3", async () => {
+      const res = await oapiForModel(
+        "Pet",
+        `
+        @name("XmlPet")
+        model Pet {
+          @name("ItemsTags")
+          @unwrapped
+          tags: Tag[];
+        }
+
+        @name("XmlTag")
+        model Tag {
+          name: string;
+        }`
+      );
+
+      deepStrictEqual(res.schemas.Tag, {
+        type: "object",
+        properties: {
+          name: {
+            type: "string",
+          },
+        },
+        required: ["name"],
+        xml: {
+          name: "XmlTag",
+        },
+      });
+
+      deepStrictEqual(res.schemas.Pet, {
+        type: "object",
+        properties: {
+          tags: {
+            type: "array",
+            xml: {
+              name: "ItemsTags",
+            },
+            items: {
+              $ref: "#/components/schemas/Tag",
+            },
+          },
+        },
+        required: ["tags"],
+        xml: {
+          name: "XmlPet",
+        },
+      });
+    });
+    it("Scenario 2.4", async () => {
+      const res = await oapiForModel(
+        "Pet",
+        `
+        @name("XmlPet")
+        model Pet {
+          @name("ItemsTags")
+          tags: Tag[];
+        }
+
+        @name("XmlTag")
+        model Tag {
+          name: string;
+        }`
+      );
+
+      deepStrictEqual(res.schemas.Tag, {
+        type: "object",
+        properties: {
+          name: {
+            type: "string",
+          },
+        },
+        required: ["name"],
+        xml: {
+          name: "XmlTag",
+        },
+      });
+
+      deepStrictEqual(res.schemas.Pet, {
+        type: "object",
+        properties: {
+          tags: {
+            type: "array",
+            xml: {
+              name: "ItemsTags",
+              wrapped: true,
+            },
+            items: {
+              $ref: "#/components/schemas/Tag",
+            },
+          },
+        },
+        required: ["tags"],
+        xml: {
+          name: "XmlPet",
+        },
+      });
+    });
+  });
+
+  describe("Nested models", () => {
+    it("Scenario 3.1", async () => {
```

Comment:
please don't name thsoe with the scenarios names for some separate doc but describe what it is testing